sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala

```diff
@@ -47,10 +47,14 @@ class MicroBatchExecution(
     triggerClock, plan.outputMode, plan.deleteCheckpointOnStop) {
 
   @volatile protected var sources: Seq[SparkDataStream] = Seq.empty
+  @volatile protected var wrappedSources: MutableMap[
+    SparkDataStream, FakeLatestOffsetSupportsTriggerAvailableNow] = new
+      MutableHashMap[SparkDataStream, FakeLatestOffsetSupportsTriggerAvailableNow]()
 
   private val triggerExecutor = trigger match {
     case t: ProcessingTimeTrigger => ProcessingTimeExecutor(t, triggerClock)
-    case OneTimeTrigger => OneTimeExecutor()
+    case OneTimeTrigger => OneBatchExecutor()
```

Comment:
It'd be better to have consistent name rule between trigger and executor. `MultiBatchExecutor` doesn't represent the semantic from its name.

Reply:
`MultiBatchExecutor` is really just running `batchRunner` multiple times. The semantic of processing all available data is implemented in `MicroBatchExecution`.

Reply:
Ah OK. I agree that the executor can be separated from the trigger and be generalized by itself.

Reply:
uber nit: `SingleBatchExecutor`, while is longer better fits with `MultiBatchExecutor`